src/index.ts

```diff
@@ -9,65 +10,145 @@ class VPCPlugin {
+    const config = this.serverless.service.custom;
+    if (!config.vpcDiscovery) {
+      config.vpcDiscovery = config.vpc;
+      Globals.logWarning(
+        "The `vpc` option of `custom` config is deprecated and will be removed in the future. " +
+        "Please use `vpcDiscovery` option instead."
+      );
+    }
+    const vpc = config && config.vpcDiscovery;
     // Checks if the serverless file is setup correctly
-    if (!vpc || vpc.vpcName == null || vpc.subnetNames == null || vpc.securityGroupNames == null) {
-      throw new Error("Serverless file is not configured correctly. Please see README for proper setup.")
+    if (!vpc || vpc.vpcName == null || (vpc.subnetNames == null && vpc.securityGroupNames == null)) {
+      throw new Error(
+        "Serverless file is not configured correctly. " +
+        "You must specify the vpcName and at least one of subnetNames or securityGroupNames. " +
+        "Please see README for proper setup."
+      );
     }
   }
 
   /**
    * Setup AWS resources
    */
   public initAWSResources (): void {
-    const credentials = this.serverless.providers.aws.getCredentials()
-    credentials.region = this.serverless.providers.aws.getRegion()
-    this.ec2Wrapper = new EC2Wrapper(credentials)
+    const credentials = this.serverless.providers.aws.getCredentials();
+    credentials.region = this.serverless.providers.aws.getRegion();
+    this.ec2Wrapper = new EC2Wrapper(credentials);
   }
 
   /**
-   * Gets the desired vpc with the designated subnets and security groups
-   * that were set in serverless config file
+   * Updates functions vpc config
    * @returns {Promise<object>}
    */
-  public async updateVpcConfig (): Promise<object> {
-    this.serverless.cli.log("Updating VPC config...")
+  public async updateFunctionsVpcConfig (): Promise<object> {
+    Globals.logInfo("Updating VPC config...");
+    const service = this.serverless.service;
+
+    // Sets the serverless's vpc config
+    if (service.functions) {
+      // get basic VPC config
+      const basicVPCDiscovery = await this.getVpcConfig(service.custom.vpcDiscovery);
+      // loop through the functions and update VPC config
+      for (const fName of Object.keys(service.functions)) {
+        const f = service.functions[fName];
+        const fVPCDiscovery = f.vpcDiscovery;
+
+        if (typeof fVPCDiscovery === "boolean" && !fVPCDiscovery) {
+          // skip vpc setup for `vpcDiscovery=false` option
+          Globals.logInfo(`Skipping VPC config for the function '${fName}'`);
+          continue;
+        }
+
+        const vpcDiscovery = basicVPCDiscovery;
+        // check vpcDiscovery config
+        if (fVPCDiscovery) {
+          const noSubnetsAndGroups = fVPCDiscovery.subnetNames == null && fVPCDiscovery.securityGroupNames == null;
+          // validate vpcDiscovery config options
+          if (fVPCDiscovery.vpcName == null || noSubnetsAndGroups) {
+            Globals.logWarning(
```

Comment:
Should there be a `continue` here so that we skip the function in this case?

Reply:
Hard to say. We need to have some feedback for this I think

My think is
If we have specified some config for this function, in general, this function should have VPC config constraints
If function config is not correct then inherit basic config

Do you think we should skip it?